includes/gateway/class-omise-payment-alipay.php

```diff
@@ -18,7 +18,7 @@ public function __construct() {
 			$this->description          = $this->get_option( 'description' );
 			$this->restricted_countries = array( 'TH' );
 
-			add_action( 'woocommerce_api_' . $this->id . '_callback', array( $this, 'callback' ) );
+			add_action( 'woocommerce_api_' . $this->id . '_callback', array( new Omise_Callback, 'execute' ) );
```

Comment:
Seems to be no point to instantiating an object here. The 'callback' object has no state and is merely being used as a holder for some functions. Would probably be better to not instantiate an object, and just use static class methods

Reply:
IIRC, I was trying static class but `add_action()` does not accept for a static class here.

Reply:
@jonrandy  Still researching, but wonder if there is any downside here by initiating a new object?

Reply:
@jonrandy another approach I'm thinking to go is to turn this `Omise_Callback` class into singleton class. I think it will be better than change it to bunch of static methods as it would give the code more flexibility and extensible ability in case if there is any more complex logic that 3rd-party want to implement by themselves. (however, not in this PR. But that extensible can come in the future PR if needed)